src/Filters/design.jl

```diff
@@ -35,8 +37,10 @@ Butterworth(n::Integer) = Butterworth(Float64, n)
 # Chebyshev type I and II prototypes
 #
 
-function chebyshev_poles(T::Type, n::Integer, ε::Real)
-    p = zeros(Complex{T}, n)
+function chebyshev_poles(::Type{T}, n::Integer, ε::Real) where {T<:Real}
+    CT = Complex{T}
+
+    p = zeros(CT, n)
```

Comment:
This seems to be the only use of `CT`. Maybe not worth introducing the extra variable?

Also in `Chebyshev1` and `Chebyshev2` below.

Reply:
It was boilerplate I introduced from `Elliptic` and I carried through.
Though I would like to point out that it is trivially optimized out.

Reply:
It definitely doesn't affect the generated code, but it makes the code more indirect and therefore harder to follow.